src/DynamoCore/Graph/ModelBase.cs

```diff
@@ -13,7 +13,7 @@ namespace Dynamo.Graph
     /// <summary>
     /// SaveContext represents several contexts, in which node can be serialized/deserialized.
     /// </summary>
-    public enum SaveContext { File, Copy, Undo, Preset, None };
+    public enum SaveContext { Save, SaveAs, File, Copy, Undo, Preset, None };
```

Comment:
these need to go at the end of the enum - otherwise this is an API break.

Reply:
I wonder if `File` is used already to mean both save and saveas?

Reply:
@mjkkirschner i can find `File` being used in these locations, seems to be mostly used with Xml stuff. Not sure what `File` represents though?

https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/NodeGraph.cs#L61
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/NodeGraph.cs#L130
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/NodeGraph.cs#L149
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/NodeGraph.cs#L187
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Models/RecordableCommands.cs#L789
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Models/DynamoModelCommands.cs#L151
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs#L1597

Reply:
thanks for digging these up, it appears to represent any save (or loading in some cases) action being done from files - IE load some nodes from a file or save some nodes - as opposed to undo redo etc because those all used to use xml serialization and deserialization.


It's not great that we now have File and Save,SaveAs since these now overlap and it will not be clear to anyone using this Enum in the future which to use - I think the simplest solution is just to obsolete `File` and replace it with `Save` ... anywhere we would have checked for `File` we can check for `File | Save | SaveAs` - what do you think?

Reply:
Yep that makes sense!

Just made a new update with these changes.